chain/chaintest/block.go

```diff
@@ -104,3 +111,270 @@ func GenerateTestExecutedBlocks(
 	}
 	return executedBlocks
 }
+
+// TXGenerator is a function that generates a valid TX that contains
+// [actions]. [factory] will be the signer of the TX.
+type TxGenerator func(actions []chain.Action, factory chain.AuthFactory) (*chain.Transaction, error)
```

Comment:
This seems to be more of a constructor than a generator. Generator implies that it generates some type of sequence whereas this is used to combine `chain.Base`, `[]chain.Action`, and `chain.AuthFactory` to create a valid HyperSDK transaction instance.

Reply:
Would it suffice instead to allow users to define a function that will provide `chain.Base` to be used at a given point in time and then have a single definition of how to combine these together? I think that would suffice and make it more clear what the caller has the opportunity to override.